rust/gui-client/src-tauri/Cargo.toml

```diff
@@ -31,10 +31,12 @@ secrecy = { workspace = true }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 subtle = "2.5.0"
-# `notification` is not needed on Windows, but if we remove it, Tauri's bundler will just add it back. #6597
-tauri = { version = "1.7.1", features = ["dialog", "icon-png", "notification", "shell-open-api", "system-tray"] }
-tauri-runtime = "0.14.5"
-tauri-utils = "1.6.0"
+tauri = { version = "2.0.3", features = ["tray-icon", "image-png"] }
+tauri-plugin-dialog = "2"
+tauri-plugin-notification = "2"
+tauri-plugin-shell = "2"
```

Comment:
Let's specify these version numbers in full too. Or did we test that `2.0.0` (which is the minimally supported one for this constraint) also works?

Reply:
Will do. It's bizarre that so many big projects (serde, Tauri, etc) want this stylish "1" and "2" junk when writing the whole number is totally harmless :/